tests/test-autover.php

```diff
@@ -0,0 +1,225 @@
+				'http://test.com/test.js?fizz=buzz&foo=bar&ver=9999'
+			],
+			[
+				'http://test.com/test.js?ver=1234&foo=bar',
+				'http://test.com/test.js?foo=bar&ver=9999'
+			],
+			[
+				'http://test.com/test.js?ver=1234',
+				'http://test.com/test.js?ver=9999'
+			],
+			[
+				'//test.com/test.js?ver=1234',
+				'//test.com/test.js?ver=9999'
+			],
+			[
+				'/test.js?ver=1234',
+				'/test.js?ver=9999'
+			],
+			[
+				'test.js?ver=1234',
+				'test.js?ver=9999'
+			],
+			[
+				'http://test.com/test+1.js',
+				'http://test.com/test+1.js?ver=9999'
+			],
+			[
+				'http://test.com/test.js?v=1234',
+				'http://test.com/test.js?ver=9999'
+			],
+			[
+				'http://user:password@test.com/test/test.css?ver=1234&foo=bar#fragment',
+				'http://user:password@test.com/test/test.css?foo=bar&ver=9999#fragment'
+			],
+			[
+				'://test.com/test.css?ver=1234&foo=bar#fragment',
+				'://test.com/test.css?foo=bar&ver=9999#fragment'
+			],
+			[
+				'//test.com/test.css?fizz=buzz&ver=1234&foo=bar#fragment',
+				'//test.com/test.css?fizz=buzz&foo=bar&ver=9999#fragment'
+			],
+			[
+				'https://test.com/test.css?fizz=buzz&ver=1234&foo=bar#fragment',
+				'https://test.com/test.css?fizz=buzz&foo=bar&ver=9999#fragment'
+			],
+			[
+				'http://test.com/test.css?fizz=buzz&ver=1234&foo=bar#fragment',
+				'http://test.com/test.css?fizz=buzz&foo=bar&ver=9999#fragment'
+			],
+			[
+				'http://test.com/test.css?fizz=buzz&ver=1234&foo=bar',
+				'http://test.com/test.css?fizz=buzz&foo=bar&ver=9999'
+			],
+			[
+				'http://test.com/test.css?ver=1234&foo=bar',
+				'http://test.com/test.css?foo=bar&ver=9999'
+			],
+			[
+				'http://test.com/test.css?ver=1234',
+				'http://test.com/test.css?ver=9999'
+			],
+			[
+				'//test.com/test.css?ver=1234',
+				'//test.com/test.css?ver=9999'
+			],
+			[
+				'/test.css?ver=1234',
+				'/test.css?ver=9999'
+			],
+			[
+				'test.css?ver=1234',
+				'test.css?ver=9999'
+			],
+			[
+				'http://test.com/test.css?v=1234',
+				'http://test.com/test.css?ver=9999'
+			],
+			[
+				'http://test.com/test+1.css',
+				'http://test.com/test+1.css?ver=9999'
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider data_css_or_js
+	 */
+	public function test_version_changed_if_css_or_js_file( $url, $expected ) {
+		redefine( 'filemtime', always( 9999 ) );
+		redefine( 'is_file', always( true ) );
+
+		$filtered_url = autover_version_filter( $url );
+
+		$this->assertSame( $expected, $filtered_url );
+	}
+
+	public function data_url_to_path() {
+		return [
+			[ 'http://test.com/test_folder/test.css', ABSPATH . 'test_folder/test.css' ],
+			[ 'http://test.com/test+1.css', ABSPATH . 'test 1.css' ],
+		];
+	}
+
+	/**
+	 * @dataProvider data_url_to_path
+	 */
+	public function test_version_acquisition_from_proper_file( $url, $expected_path ) {
+		$filemtime_called = false;
+		redefine( 'filemtime', function ( $file_path ) use ( $expected_path, &$filemtime_called ) {
+			$this->assertSame( $expected_path, $file_path );
```

Comment:
If you need to assert a variable that doesn't get back to you, then just make another flag like
```php
redefine( 'filemtime', function ( $file_path ) use ( $expected_path, &$received_path, &$filemtime_called ) {
    $received_path = $file_path;
    $filtemtime_called = true;

    return 9999;
} );

// blah blah

$this->assertSame( $expected_path, $received_path );
```